src/lib/runtime_config/runtime_config.ml

```diff
@@ -165,12 +165,88 @@ module Proof_keys = struct
              'level' to contain a string"
   end
 
-  type t = {level: Level.t option [@default None]} [@@deriving yojson]
+  module Transaction_capacity = struct
+    type t = Log_2 of int | Txns_per_second_x10 of int
+
+    let to_yojson = function
+      | Log_2 i ->
+          `Assoc [("2_to_the", `Int i)]
+      | Txns_per_second_x10 i ->
+          `Assoc [("txns_per_second_x10", `Int i)]
+
+    let of_yojson json =
+      match
+        yojson_strip_fields ~fields:[|"2_to_the"; "txns_per_second_x10"|] json
```

Comment:
Would `~keep_fields` or `keep` be a better param name than `~fields`? When I first saw this, I thought those were the fields you wanted to discard.